### ISSUE TITLE
Add FlatMapped/Flatten

### ISSUE THREAD

**Issue by krzyk** [yegor256/cactoos#1055]
I would like to have a way to turn an Iterable/Iterator/etc of Iterator/Iterable/etc into a flat Iterable/Iterator.
Basically I have something like this:
```
List<List<String>> listOfLists = new ListOf<>(new ListOf<>("one", "two"), new ListOf<>("three", "four"));
```

and I would like to be able to do:
```
List<String> list = new Flatten<>(listOfLists);
System.out.println(list);
```
and as a result I would like to see:
```
[one, two, three, four]
```

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C63314D6Z) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by llorllale** [comment]
@krzyk have you seen [this](https://stackoverflow.com/a/54465910/1623885)?

**Comment by krzyk** [comment]
@llorllale Yeah, it was asked there after I submitted the issue. But I think this should be renamed or something, `Joined` is not the first name that comes to my mind when I want to do this, but maybe I'm spoiled by streams :)

**Comment by llorllale** [comment]
@krzyk I think you *are* spoiled by streams :). `Joined` in this case is an `Iterable`, and it joins/concatenates other iterables.

The trick lies in the use of `Mapped` as seen in the SO answer above. So really, what is `flatMap` if not a composition of `Mapped` and `Joined`?

**Comment by llorllale** [comment]
@krzyk "flatMap" is the typical name for what we're talking about here

**Comment by krzyk** [comment]
@llorllale Not exactly, we are talking about `flat` in `flatMap` :)

`flatMap` is the typical name, and yet we have `Joined` and `Mapped` that's why I think `Flatted` and `Mapped` would be better names - easier to find the right classes.

**Comment by llorllale** [comment]
@krzyk I disagree on all counts.

The bottom line is that `cactoos` already provides elegant building blocks for what you want to do.

Can we close this?

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#1055` is not assigned, can't get performer

**Comment by 0crat** [comment]
This job is not in scope
